### PR TITLE
Inhibit warning messages in tiflash-proxy

### LINKED ISSUE #4616: Inhibit warning messages while building tiflash-proxy
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Meet waring like this while using latest clang in macos.
```
  /.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/651a5c0/librocksdb_sys/libtitan_sys/titan/src/db_impl.cc:100:14: error: variable 'file_size' set but not used [-Werror,-Wunused-but-set-variable]
      uint64_t file_size = 0;
               ^
  1 error generated.
  make[5]: *** [CMakeFiles/titan.dir/src/db_impl.cc.o] Error 1
  make[5]: *** Waiting for unfinished jobs....
  make[4]: *** [CMakeFiles/titan.dir/all] Error 2
  make[3]: *** [CMakeFiles/titan.dir/rule] Error 2
  make[2]: *** [titan] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2
```

### 4. What is your TiFlash version? (Required)

master

### PR DESCRIPTION
**Pull request by solotzg**
### What problem does this PR solve?

Issue Number: close #4616

since #3995

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: contrib/tiflash-proxy-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -18,10 +20,12 @@ if(TIFLASH_LLVM_TOOLCHAIN AND USE_LIBCXX)
     if(LINKER_NAME)
         set(TIFLASH_RUSTFLAGS "-C link-arg=-fuse-ld=${LINKER_NAME} ${TIFLASH_RUSTFLAGS}")
     endif()
-    set(TIFLASH_RUST_ENV CXXSTDLIB=c++ CMAKE=${CMAKE_COMMAND} CFLAGS=-w CXXFLAGS=-w RUSTFLAGS=${TIFLASH_RUSTFLAGS} PROTOC=${Protobuf_PROTOC_EXECUTABLE} PROTOC_INCLUDE=${Protobuf_INCLUDE_DIR})
-    message(STATUS "enforce LLVM toolchain for rust: ${TIFLASH_RUST_ENV}")
+    set(TIFLASH_RUST_ENV CXXSTDLIB=c++ ${TIFLASH_RUST_ENV} RUSTFLAGS=${TIFLASH_RUSTFLAGS} PROTOC=${Protobuf_PROTOC_EXECUTABLE} PROTOC_INCLUDE=${Protobuf_INCLUDE_DIR})
```
Comment (solotzg) [inline contrib/tiflash-proxy-cmake/CMakeLists.txt:23]:
```suggestion
    set(TIFLASH_RUST_ENV CC=${CMAKE_C_COMPILER} CXX=${CMAKE_CXX_COMPILER} CXXSTDLIB=c++ ${TIFLASH_RUST_ENV} RUSTFLAGS=${TIFLASH_RUSTFLAGS} PROTOC=${Protobuf_PROTOC_EXECUTABLE} PROTOC_INCLUDE=${Protobuf_INCLUDE_DIR})
```

How about using same compiler like tiflash? @SchrodingerZhu 
If not set, proxy may use default cpp compiler `cc (GCC) 4.8.5`.

Comment (SchrodingerZhu) [reply contrib/tiflash-proxy-cmake/CMakeLists.txt:23]: it is surely a better idea

Comment (SchrodingerZhu) [reply contrib/tiflash-proxy-cmake/CMakeLists.txt:23]: we can also set CMAKE to be the current executable.

Comment (SchrodingerZhu) [reply contrib/tiflash-proxy-cmake/CMakeLists.txt:23]: maybe we can use sth like $CMAKE_ROOT/bin/cmake. not quite sure

Comment (SchrodingerZhu) [reply contrib/tiflash-proxy-cmake/CMakeLists.txt:23]: for release pipeline, CXX/CC is set globally.

**Review by SchrodingerZhu — APPROVED** [review]

**Review by zanmato1984 — APPROVED** [review]
LGTM

**Pull request merged**